### PR TITLE
Revert "Index stats enhancement: creation date and tier_preference (#116339)"

### PR DESCRIPTION
**Pull request by DaveCTurner**
This reverts commit e0af1238fcdf44ebe838b1da188c8fb46a915ea1.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java [hunk 1]
```diff
@@ -202,12 +184,14 @@ public CommonStats getPrimaries() {
...
             out.writeMap(indexHealthMap, StreamOutput::writeWriteable);
             out.writeMap(indexStateMap, StreamOutput::writeWriteable);
-            out.writeMap(indexTierPreferenceMap, StreamOutput::writeStringCollection);
-            out.writeMap(indexCreationDateMap, StreamOutput::writeLong);
-
+        } else if (out.getTransportVersion().onOrAfter(TransportVersions.INDEX_STATS_ADDITIONAL_FIELDS)) {
+            out.writeMap(indexHealthMap, StreamOutput::writeWriteable);
+            out.writeMap(indexStateMap, StreamOutput::writeWriteable);
+            out.writeMap(Map.of(), StreamOutput::writeStringCollection);
+            out.writeMap(Map.of(), StreamOutput::writeLong);
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java:194]: Should we add some kind of check to be able to _eventually_ remove these?

File: server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java [hunk 2]
```diff
@@ -237,12 +221,6 @@ protected Iterator<ToXContent> customXContentChunks(ToXContent.Params params) {
                             if (indexStats.getState() != null) {
                                 builder.field("status", indexStats.getState().toString().toLowerCase(Locale.ROOT));
                             }
-                            if (indexStats.getTierPreference() != null) {
-                                builder.field("tier_preference", indexStats.getTierPreference());
-                            }
-                            if (indexStats.getCreationDate() != null) {
-                                builder.field("creation_date", indexStats.getCreationDate());
-                            }
```
Comment (DaveCTurner) [inline server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java:245]: I believe this is ok, I don't think any clients could have been expecting these fields yet and they were optional anyway so we can remove them without ceremony.

**Review by dakrone — APPROVED** [review]
LGTM, how will we eventually be able to remove the serialization?

File: server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java (hunk 1 above)
Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java:194]: Yeah eventually all these transport versions get collapsed into a release-version-named one (see e.g. `org.elasticsearch.TransportVersions#V_8_15_2`) and then this branch will be dead code:

```
        if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_16_x)) {
            indexHealthMap = in.readMap(ClusterHealthStatus::readFrom);
            indexStateMap = in.readMap(IndexMetadata.State::readFrom);
        } else if (in.getTransportVersion().onOrAfter(TransportVersions.V_8_16_x)) {
            // same condition as the previous if so cannot get here
```

**Review by thecoop — APPROVED** [review]

**Pull request merged**